File: GetInstantPrice.java
package apis;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.net.URL;
import java.util.ArrayList;

public class GetInstantPrice {
	
	public static ArrayList<String> list = new ArrayList<String>();
	public static void readList(){
		try {
		String listAddr = "C:\\Users\\jeanus\\Desktop\\list.txt";
		File f=new File(listAddr);    
        InputStreamReader isr1=new InputStreamReader(new FileInputStream(f));  
        BufferedReader br=new BufferedReader(isr1);         
    	String s;
		while ((s = br.readLine()) != null) {
			if(s.trim().startsWith("6")){
				list.add("sh"+s.trim());
			}
			if(s.trim().startsWith("0")){
				list.add("sz"+s.trim());
			}
		}
		br.close();
		}catch (Exception e) {
			e.printStackTrace();
		}
	}
	public static void main(String[] args){
		   
	        try { 
	        	readList();
	            for(String symbol : list){
	            	String res = "";   
		        	URL url = new URL("http://hq.sinajs.cn/list="+symbol);  
		            java.net.HttpURLConnection conn = (java.net.HttpURLConnection)url.openConnection();  
		            conn.setDoOutput(true);  
		            conn.setRequestMethod("POST");  
		            java.io.BufferedReader in = new java.io.BufferedReader(new java.io.InputStreamReader(conn.getInputStream(),"UTF-8"));  
		            String line;  
		            while ((line = in.readLine()) != null) {  
		                res += line;  
		            }  
		            in.close();  
		            System.out.println( symbol.substring(2)+"\t"+res.split(",")[3]);  
	            }
	        } catch (Exception e) {  
	            System.out.println("error in wapaction,and e is " + e.getMessage());  
	        }  	     
	       
		
	}

}

File: GetInstantPrice.py
# Hello World program in Python
    
# -*- coding: UTF-8 -*-
import sys, urllib

stocks= ["600271",
"600067",
"000729",
"002798",
"000517",
"002645",
"600781",
"000928",
"002314",
"603123",
"600285",
"000528",
"600048",
"002415",
"000048",
"000585",
"002351",
"002613",
"002383",
"600459",
"600608",
"600051",
"002047",
"002042",
"600195",
"600261",
"000608",
"002155",
"600615",
"603021",
"600561",
"002173",
"600988",
"000995",
"600392",
"002601",
"002371",
"000976",
"601368",
"600083",
"002719",
"000839",
"600259",
"002392",
"000525",
"002193"]

fileHandle = open ( 'test.txt', 'w' ) 
for stock in stocks :
	if stock.startswith("6"):
		stock = "sh"+stock
	if stock.startswith("0"):
		stock = "sz"+stock
	url = "http://hq.sinajs.cn/list="+stock
	wp = urllib.urlopen(url)
	content = str(wp.read())
	fileHandle.write(stock[2:]+"\t"+content.split(",")[3]+"\n")
fileHandle.close()
